apps/designer/app/designer/features/style-panel/shared/style-info.ts

```diff
@@ -0,0 +1,38 @@
+import type { Style, StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import { useMemo } from "react";
+
+export type StylePropertyInfo = {
+  source: "local" | "external";
```

Comment:
the opposite of local would be remote

Reply:
or otherwise it would be internal/external

Reply:
The problem with "generalizing" the source type early on like you do here is that if you wanted to know a more specific source (inherited/cascaded etc) somewhere later, closer to the edge - you can't.

I would much rather have the source very presize on the datatype level and use a utility function where you need to use it that would convert it to the simplistic type like local/remote.

This way you would have both - presize data and generalized type.